check_plugins_location.sh

```diff
@@ -8,39 +8,24 @@
 # SPDX-License-Identifier: EPL-2.0
 #
 
-## browse all plugin directories which names goes as follow
-##  <plugin id>/<plugin version>
-## and check that directory names match with id/version in meta.yaml files
-cd plugins
-for d in */ ; do
-  ID_DIR_NAME=${d%/}
-  cd $d
+set -e
 
-  for VERSION_DIR_NAME in */ ; do
-    # Remove trailing slash
-    VERSION_DIR_NAME=${VERSION_DIR_NAME%/}
-    cd $VERSION_DIR_NAME
+source ./util.sh
 
-    ID_YAML=$(yq r meta.yaml id | sed 's/^"\(.*\)"$/\1/')
-    if [[ "$ID_YAML" != "$ID_DIR_NAME" ]];then
-      echo "!!! ID mismatch in plugin '${ID_DIR_NAME}/${VERSION_DIR_NAME}':"
-      echo "!!!   id in meta.yaml: '${ID_YAML}'"
-      echo "!!!   id directory name: '${ID_DIR_NAME}' "
-      FOUND=true
-    fi
+declare -a arr=(`find . -name "meta.yaml"`)
+for i in "${arr[@]}"
+do
+    plugin_id=$(evaluate_plugin_id $i)
+    plugin_publisher=$(evaluate_plugin_publisher $i)
 
-    VERSION_YAML=$(yq r meta.yaml version | sed 's/^"\(.*\)"$/\1/')
-    if [[ "$VERSION_YAML" != "$VERSION_DIR_NAME" ]];then
-      echo "!!! Version mismatch in plugin '${ID_DIR_NAME}/${VERSION_DIR_NAME}':"
-      echo "!!!   version in meta.yaml: '${VERSION_YAML}'"
-      echo "!!!   version directory name: '${VERSION_DIR_NAME}' "
+    expected_path_v1="./plugins/${plugin_id}/meta.yaml"
+    expected_path_v2="./v2/plugins/${plugin_id}/meta.yaml"
+    if [[ "${expected_path_v1}" != "$i"  && "${expected_path_v2}" != "$i" ]]; then
+      echo "!!! Location mismatch in plugin '${plugin_id}':"
+      echo "!!!   Expected location: '${expected_path}' or ${alternative_expected_path}"
```

Comment:
what is alternative_expected_path? I don't see it being assigned anywhere
also missing escaping quotes

Reply:
You are looking at outdated diff

Reply:
Sorry, my bad, I was on a wrong branch. I'll remove that. Thanks for catching